### ISSUE TITLE
False positive TypeDoesNotContainNull on array_pop function

### ISSUE THREAD

**Issue by ste93cry** [vimeo/psalm#3040]
I would expect the following code to be valid as the `array_pop` function can indeed return `null`. However, the following error is thrown: `TypeDoesNotContainNull - string does not contain null`

```php
$foo = explode('...', '...');
$bar = array_pop($foo);

if (null === $bar) {
    // ...
}
```

https://psalm.dev/r/f45f448eee

**Comment by weirdan** [comment]
It returns empty string in this case: https://3v4l.org/EUEn4

**Comment by weirdan** [comment]
`explode()` returns non empty list of strings. First `array_pop()` on that list will never return null, and Psalm tells you that. Second `array_pop()` may return null, and Psalm allows that: https://psalm.dev/r/5d98db7a43
